diff --git a/sites/all/modules/contrib/brightcove/brightcove.module b/sites/all/modules/contrib/brightcove/brightcove.module
index c8bbadf..905c822 100644
--- a/sites/all/modules/contrib/brightcove/brightcove.module
+++ b/sites/all/modules/contrib/brightcove/brightcove.module
@@ -36,7 +36,7 @@ function brightcove_menu() {
     'title' => 'Brightcove settings',
     'page callback' => 'drupal_get_form',
     'page arguments' => array('brightcove_admin_settings'),
-    'type' => MENU_LOCAL_TASK,
+    'type' => MENU_NORMAL_ITEM,
     'access arguments' => array('administer brightcove settings'),
     'file' => 'brightcove.admin.inc',
   );
